File: src/main/java/org/redrune/utility/game/entity/actor/player/ChatMessage.java
package org.redrune.utility.game.entity.actor.player;

import org.redrune.utility.functions.Misc;

public class ChatMessage {
	
	private final String message;
	
	private String filteredMessage;
	
	public ChatMessage(String message) {
		if (!(this instanceof QuickChatMessage)) {
			filteredMessage = Censor.getFilteredMessage(message);
			this.message = Misc.fixChatMessage(message);
		} else {
			this.message = message;
		}
	}
	
	public String getMessage() {
		return message;
	}
	
	public String getMessage(boolean filtered) {
		if (this instanceof QuickChatMessage) {
			return message;
		}
		return filtered ? filteredMessage : message;
	}
	
}

File: src/main/kotlin/org/redrune/utility/game/entity/actor/player/ChatMessage.kt
package org.redrune.utility.game.entity.actor.player

import org.redrune.utility.functions.Misc
import org.redrune.utility.game.entity.actor.player.Censor.getFilteredMessage

open class ChatMessage(message: String?) {

    @JvmField
    var message: String? = null
    private var filteredMessage: String? = null

    init {
        if (this !is QuickChatMessage) {
            filteredMessage = getFilteredMessage(message!!)
            this.message = Misc.fixChatMessage(message)
        } else {
            this.message = message
        }
    }

    fun getMessage(filtered: Boolean): String? {
        if (this is QuickChatMessage) {
            return message
        }
        return if (filtered) filteredMessage else message
    }
}